sdk/servicebus/Microsoft.Azure.ServiceBus/src/ServiceBusConnectionStringBuilder.cs

```diff
@@ -409,7 +410,11 @@ void ParseConnectionString(string connectionString)
                 }
                 else if (key.Equals(AuthenticationConfigName, StringComparison.OrdinalIgnoreCase))
                 {
-                    this.Authentication = value;
+                    value = value.Replace(" ", string.Empty);
+                    if (!Enum.TryParse(value, true, out this.authType))
```

Comment:
We should consider adding an `Enum.IsDefined` check here as well, since the parse will succeed for numeric values that are not part of the enumeration, which `IsDefined` would catch.   Since `AuthenticationType.None` is the 0 value for the enum, there's no need to assign it on failure.   The `out` from the `TryParse` will set that.   

I'd recommend refactoring to something like:

```csharp
var _ = ((Enum.IsDefined(typeof(AuthenticationType), value) && Enum.TryParse(value, true, out this.authType)));
  
// At this point, this.authType will either be the parsed value or the default (AuthenticationType.None) - unless you explicitly assigned it elsewhere that I've overlooked.
```

Reply:
For fun and illustration:

```csharp
var result = Enum.TryParse<AuthenticationType>("999", true, out var authType);
```
At this point:
  - `result == true`
  - `authype == 999`

Any numeric string will work, the same as a straight cast: `(AuthenticationType)888`

Reply:
`IsDefined` doesn't seem to be case insensitive, I'm not sure if we want to do that

---
In reply to: [295835374](https://github.com/Azure/azure-sdk-for-net/pull/6637#discussion_r295835374) [](ancestors = 295835374)

Reply:
In that case, maybe disallow numeric values by ensuring a `!int.TryParse(out var _)`?  That would, at least, close the int-as-a-string hole.